viper.go

```diff
@@ -1084,11 +1084,16 @@ func (v *Viper) AutomaticEnv() {
 	v.automaticEnvApplied = true
 }
 
-// SetEnvKeyReplacer sets the strings.Replacer on the viper object
+// replacer is the interface that a replacement algorithm needs to implement.
+type replacer interface {
```

Comment:
Forgive me if I'm being dumb, but shouldn't this interface be exported so someone can pass a custom implementation in to the SetEnvKeyReplacer() calls?

Reply:
I had the same idea. But [strings.replacer](https://golang.org/src/strings/replace.go) is private for some reason. I thought: "Maybe it is some kind of convention to make such interfaces to be independent". Should I fix it?

Reply:
No, it can stay unexported, noone needs to refer to this type directly. I would probably paste the interface definition in comment above the `SetEnvKeyReplacer`.

See examples for unexported interface usage in this package: https://github.com/pkg/errors

Reply:
Actually, I changed my mind. I think it should be exported after all.

(I forgot about this PR and submitted my own: #624)